File: apps/web/src/components/knowledge/document-import-policy.ts
import type { DocumentImportImagePolicy } from "@/lib/api"

export const IMAGE_POLICY_LABELS: Record<DocumentImportImagePolicy, string> = {
  keep_and_recognize: "保留图片并识别",
  text_only: "仅识别文字",
  images_only: "仅保留图片",
}

export const IMAGE_POLICY_DESCRIPTIONS: Record<DocumentImportImagePolicy, string> = {
  keep_and_recognize: "图片保留在正文中；扫描文字和截图做 OCR，照片与图表补充描述。",
  text_only: "只提取图片中的可见文字，不补图片描述，文章不插入图片或原页预览。",
  images_only: "图片保留在正文中，跳过图片分类、OCR 和描述，不消耗图片识别额度。",
}

File: apps/api/internal/taskqueue/document_import_policy.go
package taskqueue

import "strings"

const (
	ImagePolicyKeepAndRecognize = "keep_and_recognize"
	ImagePolicyTextOnly         = "text_only"
	ImagePolicyImagesOnly       = "images_only"
)

// 缺省沿用历史任务行为，保证升级后重试不会改变文章内容。
func EffectiveDocumentImagePolicy(policy string) string {
	if policy == "" {
		return ImagePolicyKeepAndRecognize
	}
	return policy
}

func ValidDocumentImagePolicy(policy string) bool {
	switch EffectiveDocumentImagePolicy(policy) {
	case ImagePolicyKeepAndRecognize, ImagePolicyTextOnly, ImagePolicyImagesOnly:
		return true
	}
	return false
}

// 仅文字策略的图片只用于 OCR 重试，不作为长期文章资源。
func DocumentImportOCRAssetKey(userID, jobID int64, token string, pageNo int32, index int) string {
	return strings.Replace(DocumentImportAssetKey(userID, jobID, token, pageNo, index), "/document-assets/", "/document-import/", 1)
}
